src/contracts/middleware/IndexRegistry.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IRegistryCoordinator.sol";
+import "../libraries/BN254.sol";
+import "forge-std/Test.sol";
+
+
+contract IndexRegistry is IIndexRegistry, Test {
+
+    IRegistryCoordinator public registryCoordinator;
+
+    // list of all unique registered operators
+    bytes32[] public globalOperatorList;
+
+    // mapping of operatorId => quorumNumber => index history of that operator
+    mapping(bytes32 => mapping(uint8 => OperatorIndex[])) public operatorIdToIndexHistory;
+    // mapping of quorumNumber => history of numbers of unique registered operators
+    mapping(uint8 => OperatorIndex[]) public totalOperatorsHistory;
```

Comment:
probably again want the mappings which return a struct to be internal (+ name prepended `_`) with an external/public getter function that properly returns a struct in memory, rather than a tuple of the struct contents

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/4f13cb5ce588cef908b9218011843ce63177706c good call

Reply:
this was resolved but not for the contract in question

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/f38b7c8b49956ccf142cc24408f02452ce80e9d3